File: macos/Sources/OpenTimer/Models/WorkPackage.swift
import Foundation

/// Un work package OpenProject, décodé depuis la représentation HAL de l'API v3.
struct WorkPackage: Identifiable, Hashable {
    let id: Int
    let subject: String
    let projectName: String?

    var href: String { "/api/v3/work_packages/\(id)" }

    var display: String {
        if let projectName, !projectName.isEmpty {
            return "#\(id) — \(subject)  ·  \(projectName)"
        }
        return "#\(id) — \(subject)"
    }
}

/// Une activité de temps autorisée pour un work package (Développement, Réunion, …).
struct Activity: Identifiable, Hashable {
    let href: String
    let name: String
    var id: String { href }
}

File: windows/src/OpenTimer.Windows/Models/WorkPackage.cs
namespace OpenTimer.Models;

/// <summary>
/// Un work package OpenProject, décodé depuis la représentation HAL de l'API v3.
/// </summary>
public sealed record WorkPackage(int Id, string Subject, string? ProjectName)
{
    public string Href => $"/api/v3/work_packages/{Id}";

    public string Display => string.IsNullOrEmpty(ProjectName)
        ? $"#{Id} — {Subject}"
        : $"#{Id} — {Subject}  ·  {ProjectName}";
}

/// <summary>
/// Une activité de temps autorisée pour un work package (Développement, Réunion, …).
/// </summary>
public sealed record Activity(string Href, string Name)
{
    public override string ToString() => Name;
}
